From 475b272c126d53f3985860422dc96b01b470423e Mon Sep 17 00:00:00 2001
From: Chunyan Liu <cyliu@suse.com>
Date: Tue, 29 Jul 2014 17:31:28 +0800
Subject: [PATCH] blockcopy: check dst = identical device

Check whether dst is the same device as source, if yes, report
error and exit.

Signed-off-by: Chunyan Liu <cyliu@suse.com>
---
 src/qemu/qemu_driver.c | 7 +++++++
 1 file changed, 7 insertions(+)

Index: libvirt-5.6.0/src/qemu/qemu_driver.c
===================================================================
--- libvirt-5.6.0.orig/src/qemu/qemu_driver.c
+++ libvirt-5.6.0/src/qemu/qemu_driver.c
@@ -18403,6 +18403,14 @@ qemuDomainBlockCopyCommon(virDomainObjPt
         goto endjob;
     }
 
+    if (STREQ_NULLABLE(realpath(disk->src->path, NULL),
+                       realpath(mirror->path, NULL))) {
+        virReportError(VIR_ERR_INVALID_ARG,
+                       _("destination '%s' is the same as disk '%s' source"),
+                       mirror->path, path);
+        goto endjob;
+    }
+
     if (qemuDomainStorageFileInit(driver, vm, mirror, NULL) < 0)
         goto endjob;
 
